### ISSUE TITLE
[Bug] RateLimitBundle will not autoload anymore in Composer v2.0

### ISSUE THREAD

**Issue by themihel** [jaytaph/RateLimitBundle#112]
### Basic information
Current installed version: `1.15.1` via packagist.

### Description
As stated in the title, the RateLimitBundle will not autoload anymore in Composer v2.0 because it does not comply with the psr-0 autoloading standard.

In our current installation we are getting the following error for different (mostly test) classes:
```
Deprecation Notice: Class Noxlogic\RateLimitBundle\Tests\Annotation\MockListener located in ./vendor/noxlogic/ratelimit-bundle/Noxlogic/RateLimitBundle/Tests/EventListener/BaseListenerTest.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///opt/sources/composer/composer/src/Composer/Autoload/ClassMapGenerator.php:201
```

The whole list in our case is:
* `Noxlogic\RateLimitBundle\Tests\Annotation\MockListener`
* `Noxlogic\RateLimitBundle\Tests\Annotation\BaseListenerTest`
* `Noxlogic\RateLimitBundle\Tests\Annotation\HeaderModificationListenerTest`
* `Noxlogic\RateLimitBundle\Tests\Annotation\OauthKeyGenerateListenerTest`
* `Noxlogic\RateLimitBundle\EventListener\Tests\MockController`
* `Noxlogic\RateLimitBundle\EventListener\Tests\RateLimitAnnotationListenerTest`
* `Noxlogic\RateLimitBundle\Tests\Annotation\CheckedRateLimitEventsTest`
* `Noxlogic\RateLimitBundle\Tests\Annotation\GenerateKeyEventsTest`
* `Noxlogic\RateLimitBundle\Tests\Annotation\RateLimitEventsTest`
* `Noxlogic\RateLimitBundle\Tests\Annotation\RateLimitInfoTest`
* `Noxlogic\RateLimitBundle\Tests\Annotation\RateLimitServiceTest`

### Possible fix
The easiest way to fix this is either to move the test classes to the respective directory or adapt the namespace.
**If you want, I can hand in a pull request. But I first want to make sure the way you want to adapt this.**

**Comment by jaytaph** [comment]
I think this should indeed be fixed. Love to see a pull request for it!

**Comment by themihel** [comment]
All right, I will probably start working on it tonight. 
My basic approach is to simply adapt the namespaces of the tests to the current folder structure.

**Comment by themihel** [comment]
I just noticed @mhor already fixed this in #105.
So you would just need to release a new version of the Bundle as the change was merged into master on 1 Apr and the last release is based on the changes of  11 feb.

Is there a new release coming up anywhere soon?

**Comment by jaytaph** [comment]
ah. I already forgot that this was done (and didn't check).

I think we can do a release for composer 2.0 compatibility. I think there is not a lot of additional code current pending so it should be no problem. @mcfedr , should we do a 1.16.0 release for this?

**Comment by jaytaph** [comment]
https://github.com/jaytaph/RateLimitBundle/compare/1.15.1..master

seems like it's just psr-0 replacement and phpredis cluster functionality. I think this is ok for a 1.16.0 release?

**Comment by jaytaph** [comment]
might as well check php 8.0 compatibility and update composer.json accordingly

**Comment by themihel** [comment]
Hey,
just wanted to ask if there are any updates on this?

Idea: Is it maybe possible to release a new version of the bundle without PHP8 support first, and push the compatibilities in a later version?

Thanks in advance!

**Comment by jaytaph** [comment]
yeah.. let's get this out of the way :) 

I can push a 1.16.0 release, that will fix the psr-0 issues.

Next, we can do a check on php8 support, and push a 1.17.0 release accordingly if this needs changes (at least the composer.json will need to be updated)

**Comment by jaytaph** [comment]
Just released 1.16.0 which should fix the composer2 issue:

```
➜  tmp php composer.phar -V
Composer version 2.0.9 2021-01-27 16:09:27

➜  tmp php composer.phar require noxlogic/ratelimit-bundle
Using version ^1.16 for noxlogic/ratelimit-bundle
./composer.json has been updated
Running composer update noxlogic/ratelimit-bundle
Loading composer repositories with package information
Updating dependencies
Lock file operations: 30 installs, 0 updates, 0 removals
  - Locking doctrine/annotations (1.11.1)
  - Locking doctrine/lexer (1.2.1)
  - Locking noxlogic/ratelimit-bundle (1.16.0)
  - Locking psr/cache (1.0.1)
  - Locking psr/container (1.0.0)
  - Locking psr/event-dispatcher (1.0.0)
  - Locking psr/log (1.1.3)
  - Locking sensio/framework-extra-bundle (v5.6.1)
  - Locking symfony/cache (v5.2.2)
  - Locking symfony/cache-contracts (v2.2.0)
  - Locking symfony/config (v5.2.2)
  - Locking symfony/dependency-injection (v5.2.2)
  - Locking symfony/deprecation-contracts (v2.2.0)
  - Locking symfony/error-handler (v5.2.2)
  - Locking symfony/event-dispatcher (v5.2.2)
  - Locking symfony/event-dispatcher-contracts (v2.2.0)
  - Locking symfony/filesystem (v5.2.2)
  - Locking symfony/finder (v5.2.2)
  - Locking symfony/framework-bundle (v5.2.2)
  - Locking symfony/http-client-contracts (v2.3.1)
  - Locking symfony/http-foundation (v5.2.2)
  - Locking symfony/http-kernel (v5.2.2)
  - Locking symfony/polyfill-ctype (v1.22.0)
  - Locking symfony/polyfill-mbstring (v1.22.0)
  - Locking symfony/polyfill-php73 (v1.22.0)
  - Locking symfony/polyfill-php80 (v1.22.0)
  - Locking symfony/routing (v5.2.2)
  - Locking symfony/service-contracts (v2.2.0)
  - Locking symfony/var-dumper (v5.2.2)
  - Locking symfony/var-exporter (v5.2.2)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 30 installs, 0 updates, 0 removals
  - Downloading noxlogic/ratelimit-bundle (1.16.0)
  - Installing doctrine/lexer (1.2.1): Extracting archive
  - Installing symfony/polyfill-php80 (v1.22.0): Extracting archive
  - Installing symfony/deprecation-contracts (v2.2.0): Extracting archive
  - Installing symfony/routing (v5.2.2): Extracting archive
  - Installing symfony/polyfill-mbstring (v1.22.0): Extracting archive
  - Installing symfony/polyfill-php73 (v1.22.0): Extracting archive
  - Installing symfony/polyfill-ctype (v1.22.0): Extracting archive
  - Installing symfony/http-foundation (v5.2.2): Extracting archive
  - Installing symfony/http-client-contracts (v2.3.1): Extracting archive
  - Installing psr/event-dispatcher (1.0.0): Extracting archive
  - Installing symfony/event-dispatcher-contracts (v2.2.0): Extracting archive
  - Installing symfony/event-dispatcher (v5.2.2): Extracting archive
  - Installing symfony/var-dumper (v5.2.2): Extracting archive
  - Installing psr/log (1.1.3): Extracting archive
  - Installing symfony/error-handler (v5.2.2): Extracting archive
  - Installing symfony/http-kernel (v5.2.2): Extracting archive
  - Installing symfony/finder (v5.2.2): Extracting archive
  - Installing symfony/filesystem (v5.2.2): Extracting archive
  - Installing psr/container (1.0.0): Extracting archive
  - Installing symfony/service-contracts (v2.2.0): Extracting archive
  - Installing symfony/dependency-injection (v5.2.2): Extracting archive
  - Installing symfony/config (v5.2.2): Extracting archive
  - Installing symfony/var-exporter (v5.2.2): Extracting archive
  - Installing psr/cache (1.0.1): Extracting archive
  - Installing symfony/cache-contracts (v2.2.0): Extracting archive
  - Installing symfony/cache (v5.2.2): Extracting archive
  - Installing symfony/framework-bundle (v5.2.2): Extracting archive
  - Installing doctrine/annotations (1.11.1): Extracting archive
  - Installing sensio/framework-extra-bundle (v5.6.1): Extracting archive
  - Installing noxlogic/ratelimit-bundle (1.16.0): Extracting archive
23 package suggestions were added by new dependencies, use `composer suggest` to see details.
Generating autoload files
23 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
```
